### PR TITLE
feat(scanner): detect Cloudflare account API tokens (cfat_ prefix)

### LINKED ISSUE #210: rule: Cloudflare cfat_ prefix
Parent: #171.

## What to change
Cloudflare `cfat_` detector + one test file.

Comment to claim it.

### PR DESCRIPTION
**Pull request by Suyoag**
## What & why

Adds detection for Cloudflare's account-owned API token format (`cfat_` prefix), part of the 2026 scannable credential rollout described in #171. This is a local-only rule (no gitleaks equivalent yet).

Closes #210

## Format

`cfat_` + 40 alphanumeric characters + an 8-character hex checksum, per Cloudflare's DLP predefined-profiles documentation. Cross-checked against a real example token in the `caddy-dns/cloudflare` provider's test suite (`cfut_...`, same body shape, different prefix) — 48 total body chars, first 40 alphanumeric, last 8 hex. Checksum is length/charset-matched only, not algorithmically validated (kept out of the regex per the parent issue's guidance).

## Type of change

- [x] New detection rule

## Testing

```
755 passed, 10 skipped
mypy src/: Success: no issues found in 30 source files
ruff check: All checks passed
bandit -r src/: clean
```

## Checklist

- [x] `pytest` passes locally
- [x] `mypy src/` is clean
- [x] No new runtime dependency
- [x] No real secrets, tokens, or raw history-file contents are committed — fixture is synthetic, split across string literals, and `.gitguardian.yaml` carries an allowlist entry for the anchor literal
- [x] New detection rule: added to `RULES`, `ROTATION_GUIDANCE`, prefilter anchor, and a fixture/test, with the regex fully bounded (`{40}`/`{8}`, no unbounded `.*`)
- [x] `--json` / non-tty output unaffected (no CLI changes)
- [x] Did not touch `force-include` in `pyproject.toml`

## Summary by CodeRabbit

- **New Features**
  - Added detection for Cloudflare account API tokens.
  - Provides rotation guidance when a matching token is found.

- **Bug Fixes**
  - Validates the expected token length and hexadecimal format.
  - Reduces false positives when token-like text appears within words or beside dashes or underscores.

- **Tests**
  - Added coverage for valid tokens, formatting requirements, boundary conditions, rotation guidance, and regex compatibility.

<details><summary><h3>Greptile Summary</h3></summary>

The PR adds local detection and rotation guidance for Cloudflare account API tokens using the `cfat_` format.
- Registers the bounded token rule, prefilter anchor, and drift metadata.
- Adds format, boundary, guidance, and regex-engine parity coverage.
- Updates the GitGuardian allowlist for the synthetic prefix literal.
</details>

<h3>Confidence Score: 5/5</h3>

The PR appears safe to merge.

No blocking failure remains.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/agentsweep/scanner.py | Adds the bounded Cloudflare account-token rule, its lossless prefilter anchor, and rotation guidance. |
| tests/test_cloudflare_account_api_token_rule.py | Covers successful detection, exact body and checksum requirements, and identifier-boundary rejection. |
| tests/test_regex_engine_parity.py | Adds a synthetic Cloudflare token fixture to existing engine and prefilter parity coverage. |
| scripts/rules_drift.py | Marks the new local rule as having no corresponding Gitleaks rule. |
| .gitguardian.yaml | Allowlists only the synthetic `cfat_` rule anchor used by the repository. |

<sub>Reviews (2): Last reviewed commit: ["fix: satisfy ruff format on split-litera..."](https://github.com/ishannaik/agent-sweep/commit/01721dab9ee7dc9fc4ad83b636dafd54623fdea4) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=55217990)</sub>

### CODE REVIEW

File: tests/test_cloudflare_account_api_token_rule.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "embedded",
+    [
+        "z" + _token(),
+        "-" + _token(),
+        _token() + "z",
+        _token() + "-",
+    ],
+)
+def test_cloudflare_account_api_token_rejects_word_and_dash_embeds(embedded: str):
+    assert scan_text(embedded) == []
```
[the remark this answers is not in the document]
Comment (Suyoag) [reply tests/test_cloudflare_account_api_token_rule.py:48]: Added — pushed underscore-prefix and underscore-suffix cases to the parametrize list in 01721da.

**Comment by Ishannaik** [comment]
Needs rebase onto current main after Tavily merge; please update.

**Comment by Ishannaik** [comment]
Reopened. This was closed as "needs rebase" without a review, and the cfat_ rule is still missing from main, so the change is wanted. It now conflicts in 5 files. Rebase on main and I will run it and merge.

**Comment by Ishannaik** [comment]
Merged as 3ce9aea. I merged main into your branch instead of rewriting it, and where the fixtures conflicted I kept main's neon and pinecone lines, so only your cfat rule, test file, and parity fixture changed. Locally: 838 passed, ruff and mypy clean, `scan_text` fires on a 40+8 token and stays silent on a non-hex checksum. That makes 211 rules.
